### DISCUSSION TITLE
Add inline parameter hints to function calls in the editor

### DISCUSSION THREAD

**Discussion by tobydjones** [godotengine/godot-proposals#10732] [Editor]
If I have a function:

```
func move_sprite(sprite_moved: bool, new_pos: Vector2, SP_name: String):
```
The function call looks like this:
```
    move_sprite(true, Vector2(0,0), "Home")
```
It’s hard to know what the parameters refer to without looking at the function.

In VSCode, it's possible to turn on inline parameter hints. This shows the parameter name as a small comment next to the passed parameter. I don't know if VSCode will do this for GDscript, this example I found is, I think, C++:

![image](https://github.com/user-attachments/assets/8e54f25c-b1d4-41b8-8537-43b79bbc6d18)

If inline parameter hints were added to the Godot editor, code would be far easier to maintain. Here's a mockup:

![inline hints](https://github.com/user-attachments/assets/8f352cec-0673-4cb0-9678-001e5ec9c728)

Please note I am not suggesting passing arguments by name.

edit:

I believe this should be possible to implement, as the editor already knows the parameters names and shows them as a popup when coding:

![image](https://github.com/user-attachments/assets/e210c962-0530-4952-9464-d59762238748)

Of course, these inline hints would have to be updated if the parameter names in the function definitions are edited.

**Comment by Meorge** [comment]
This would be great to have, both in Godot's script editor and the VS Code plugin! (I use the latter, and this feature doesn't appear for me when using GDScript).

A while back I was interested in keyword arguments for GDScript and did a little reading about their feasibility of being implemented, and understood that it would be difficult due to the way the parser was built. I'd be a little worried that functionality like this would face the same hurdles (i.e., the parser not keeping track of argument names, and thus not being able to forward them to an editor). Someone more knowledgeable on the GDScript system may be able to provide more insightful commentary here, though.

**Comment by dalexeev** [comment]
* See also #7683.

**Reply by tobydjones** [reply]
@dalexeev thank you for taking the time to read my proposal.
I'd like to point out (for anyone thinking of merging the two proposals) that the proposal you linked (though similar) is for variable _type_, whereas my proposal is for the _names_ of function parameters.

**Reply by DaloLorn** [reply]
I feel like they might be branches of a larger proposal here that has yet to be made. @mrjustaguy had me doing a little research for him (he hoped I'd open the proposal for him because I more regularly use IDEs with inlay hints), and the JetBrains IDEs have a pretty robust set of inlay hint types above and beyond what was discussed here and in #7683 *combined*.

**Comment by omniuni** [comment]
This would be INCREDIBLY useful. Also, unlike the type hints, this should not have an inference issue.

**Comment by Ariorick** [comment]
Yes please! That would be great

**Reply by Calinou** [reply]
@Ariorick Please don't bump issues without contributing significant new information. Use the :+1: reaction button on the first post instead.
